Vienna/Sources/Fetching/RefreshManager.m

```diff
@@ -750,6 +742,40 @@ -(void)finalizeFolderRefresh:(NSDictionary *)parameters
             feedLink = [NSURL URLWithString:feedLink relativeToURL:url].absoluteString;
         }
 
+        if (feedTitle != nil  && !feedTitle.vna_isBlank && [folder.name hasPrefix:[Database untitledFeedFolderName]]) {
+            // If there's an existing feed with this title, make ours unique
+            // BUGBUG: This duplicates logic in database.m so consider moving it there.
+            NSString * oldFeedTitle = feedTitle;
+            NSString * newFeedTitle = feedTitle;
+            NSUInteger index = 1;
+
+            while (([dbManager folderFromName:newFeedTitle]) != nil) {
+                newFeedTitle = [NSString stringWithFormat:@"%@ (%lu)", oldFeedTitle, (unsigned long)index++];
+            }
+
+            connectorItem.name = newFeedTitle;
+            [dbManager setName:newFeedTitle forFolder:folderId];
+        }
+        if (feedDescription != nil) {
+            [dbManager setDescription:feedDescription forFolder:folderId];
+        }
+        if (feedLink != nil) {
+            [dbManager setHomePage:feedLink forFolder:folderId];
+        }
+        // Remember the last modified date
+        if (lastModifiedString != nil && lastModifiedString.length > 0) {
+            [dbManager setLastUpdateString:lastModifiedString forFolder:folderId];
+        }
+        // Set the last update date for this folder.
+        [dbManager setLastUpdate:[NSDate date] forFolder:folderId];
```

Comment:
The last update date is displayed in the "Get Info…" window, and is useful for detecting that the feed is stale.
Therefore, we should update it only if new items are present…

Reply:
This conversation takes an interesting turn…

First, I have to make clear that, for a folder, the only role of the `lastUpdate` field is to be displayed in the "Get Info…" window.
It must not be confused with `lastUpdateString`, which is provided by the server and is used for optimizing network access.

That being said, the word "refreshed" can have slightly different meanings. In the context, I identify at least three possible nuances:
- fetched from server;
- got some article(s);
- got some _new_ article(s) (_fresh_ article(s)…)

From a user point of view, I think the most interesting meaning is the third one. Current implementation in OpenReader is aligned with this latest meaning. Cf. https://github.com/ViennaRSS/vienna-rss/blob/196afbb1aef855c68894a1b4cf03cd790fa5ce2e/Vienna/Sources/Fetching/OpenReader.m#L640-L655

I am not sure if this is what led me to modify the French translation of the "Last Refreshed" label during the development of version 3.8, in 2021 : https://github.com/ViennaRSS/vienna-rss/commit/251861e4ec0e0f00fb52cb0abfe239731e00cf70#diff-987fbfdc228b4925f39ac4eaeae776a082dc0d34c459450200654da97f34ffd2 . I replaced "Dernier rafraîchissement" (which does not really sound French) with "Dernière actualisation" (which is closer to "Last Update").

My current opinion is that we should make Vienna consistent and adopt the OpenReader behavior mentioned beforehand:
- do not set `lastUpdate` anymore on 304 HTTP status : https://github.com/ViennaRSS/vienna-rss/blob/196afbb1aef855c68894a1b4cf03cd790fa5ce2e/Vienna/Sources/Fetching/RefreshManager.m#L610-L614
- update it only if `newArticlesFromFeed`> 0

The change would constitute a first step towards issue #1638